languages/base-nodemaker.lua

```diff
@@ -0,0 +1,96 @@
+local chardata = require("char-def")
```

Comment:
On the naming of this file: why is it `languages/base-nodemaker.lua` and not e.g. `languages/nodemakers/base.lua` (as we have for other typesetters, classes, shapers, etc.)

Reply:
Good question. Because all the directories in `languages/*` are language codes and I wanted to be able to iterate them to get a list of languages we support at some level without bumping into things that were not languages.

That also means these should probably go somewhere else entirely. I'll think about what to do with supporting module types that are not part of our core module loader system. Maybe they should just be promoted to top level, I dunno.